[GPU] Fix users invalidation and output layout recalculation in  graph dump mode

sshlyapn opened the pull request:
This PR adds new `program::get_output_layout(const program_node& node)` method for accessing private field of `program_node`. That allows do not trigger output layout recalculation and users invalidation in graph dump mode by calling `node.get_output_layout(invalidate_users_if_changed=true)` function, which may affect final graph

isanghao commented:
@sshlyapn  We've added code like this: https://github.com/openvinotoolkit/openvino/pull/9196/files#diff-6483fde32714a7ae7d5dfd90c3e02e2ec1722990e992ddf8d5dc9dddb5720b22R1228

If you want to keep the invalid layout as-is, I think you need to remove that part.

sshlyapn commented:
> @sshlyapn We've added code like this: https://github.com/openvinotoolkit/openvino/pull/9196/files#diff-6483fde32714a7ae7d5dfd90c3e02e2ec1722990e992ddf8d5dc9dddb5720b22R1228
> 
> If you want to keep the invalid layout as-is, I think you need to remove that part.
> 
> JFYI @jade-cho

@isanghao thank you for comment, I removed that part in debug mode. I think it's better to dump original layout in debug mode w/o any modifications

isanghao commented on src/plugins/intel_gpu/include/intel_gpu/graph/program.hpp:
```diff
@@ -212,6 +212,7 @@ struct program {
     std::string get_implementation_info(const primitive_id& id) const;
     const graph_optimizer_info& get_optimizer_passes_info() const;
     void save_pass_info(std::string pass_name);
+    layout get_output_layout(const program_node& node) const;
```
How about changing name to avoid confusion? Now we have three versions of `get_output_layout` and that is quite a lot... What about something like `get_output_layout_no_touch`?
And it would be better to leave some comment on the usage or difference.

sshlyapn replied:
Vladimir and I decided not to add new methods to program class and not to display any layout information if output_layout is invalid

isanghao replied:
Great, I think that is better.

isanghao replied:
Could you post on the channel for this change when merged?
